src/WalkingModule/src/Module.cpp

```diff
@@ -1239,6 +1289,22 @@ bool WalkingModule::setPlannerInput(double x, double y)
 
     m_newTrajectoryRequired = true;
 
+    double treshold = 0.01;
+    // the robot is about to move
+    if(iDynTree::toEigen(m_desiredPosition).norm() > treshold)
+    {
+        m_isStancePhase = false;
+        m_isStancePhaseStarting = false;
+    }
+    else
+    {
+        // if the stance phase has not started reset the counter
+        if(!m_isStancePhaseStarting)
+            m_stancePhaseCounter = m_stancePhaseMaxCounter;
+
+        m_isStancePhaseStarting = true;
```

Comment:
Can't we create a function in the trajectory generator class asking for the phase value in a certain number of steps in the future?

Reply:
Done in https://github.com/robotology/walking-controllers/pull/41/commits/9760be0556d233e940c1faba357e1d48f3f54cf8